dashboard/app/controllers/home_controller.rb

```diff
@@ -1,13 +1,19 @@
 class HomeController < ApplicationController
   before_filter :authenticate_user!, only: :gallery_activities
 
-
   # Don't require an authenticity token on set_locale because we post to that
   # action from publicly cached page without a valid token. The worst case impact
   # is that an attacker could change a user's language if they fooled them into
   # clicking on a link.
   skip_before_action :verify_authenticity_token, :only => 'set_locale'
 
+  # Sets the client state key for the current user and redirects to params[:redirect] or
+  # the root if unspecified.
+  def update_login
```

Comment:
On the fence about this (may be fine as-is), but wondering if this name+associated route might benefit from a bit more specificity. E.g., `update_[client|frontend]_login`, `update_client_for_user`, `set_client_user`